### ISSUE TITLE
dependencies are ignored in pnpm8 if they are declared as optional dependencies at same time.

### ISSUE THREAD

**Issue by Mister-Hope** [pnpm/pnpm#6420]
### Last pnpm version that worked:

v7

### pnpm version:

All v8 versions

Here is an example package:

```json
{
   "name": "packageB",
  "dependencies": {
    "packageA": "1.0.0-beta.1",
  },
  "peerDependencies": {
    "packageA": "1.0.0-beta.1"
  },
  "peerDependenciesMeta": {
    "packageA": {
      "optional": true
    }
}
```

Package A is kind of plugin, but it's in beta stage, so if users manually adds pluginA and set something himself, he should install the same version `packageB` is using when installing `packageB`

In pnpm7, this feature is working as expected, as pnpm will give warning if the version mismatches.

However, in pnpm8, the packageA is just not installed, I am not sure but I believed  this behavior is related to https://github.com/orgs/pnpm/discussions/6329, where pnpm8 disallows a package to have same package as deps and optional peer, which differs with npm and yarn behavior.

I hope someone in official team can explain some details why pnpm choose this behvaior.

**Comment by zkochan** [comment]
`packageA` should be installed for sure. However, this `package.json` doesn't make sense because if the peer dependency is added to the `"dependencies"` field then it cannot be an optional peer dependency as it will be always installed.

**Comment by Mister-Hope** [comment]
Yes, packageA should always be installed, but a peer package A should satisfy the version.

Current behaviour is package A not installed, so this should be a bug, similar to my another one.
